pkg/resources/virtual/common/common.go

```diff
@@ -29,6 +29,11 @@ func (d *DefaultFields) GetTransform() cache.TransformFunc {
 	return d.transform
 }
 
+// GetEventTransform produces the default transformation func
+func (d *DefaultFields) GetEventTransform() cache.TransformFunc {
```

Comment:
I don't think that this fits the purpose of "DefaultFields". Default fields are the fields that are the same for all objects, but this function is new code specific to a single type. I think this needs to go in it's own package, along with the logic to change the type field.

Reply:
I've worked out something that still uses the `common` package, pulling the event-specific transformer into another file. It looks a bit hokey but it can be used to scale up to cover more types in the future.

Reply:
lgtm